components/common/SubMenu.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Link } from 'routes';
+
+export default class SubMenu extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    // bindings
+    this._onClickOutside = this.onClickOutside.bind(this);
+  }
+
+  componentDidMount() {
+    this.setEventListeners();
+  }
+
+  componentWillUnmount() {
+    this.removeEventListeners();
+  }
+
+  onClickOutside(e) {
+    const isClickOutside = this.subMenuNode &&
+      this.subMenuNode.contains &&
+      !this.subMenuNode.contains(e.target) &&
+      !this.props.parentNode.contains(e.target);
+
+    if (isClickOutside) {
+      this.props.onCloseSubMenu();
+    }
+  }
+  onClickItem(e, value) {
+    if (e) e.preventDefault();
+    this.props.onClick(value);
+  }
+
+  setEventListeners() {
+    window.addEventListener('click', this._onClickOutside);
+  }
+
+  removeEventListeners() {
+    window.removeEventListener('click', this._onClickOutside);
+  }
+
+  render() {
+    const { items, className, parent } = this.props;
+    const classNames = classnames('c-submenu', {
+      [className]: !!className
+    });
+
+    return (
+      <div className={classNames} ref={(node) => { this.subMenuNode = node; }}>
+        <div className="parent-menu"><span className="literal">{parent}</span></div>
+        <ul className="menu-list" role="menubar" aria-label={`Submenu for ${parent} section`}>
+          {items.map(item => <li key={item.value} className="menu-item" role="menuitem" tabIndex="-1">
+            <Link
+              href={{ pathname: item.pathname }}
```

Comment:
Don't use this syntax. Only the `routes.js` file know the full path. Instead, simply use:
`<Link route="explore" params={{ category: item.value } prefetch>...</Link>`

For this, the parent component will need to pass you the name of the route (`explore`) and the name of the params (`category`).